indico/cli/i18n.py

```diff
@@ -113,62 +244,246 @@ def _wrapper(**kwargs):
+    python = kwargs.get('python')
+    javascript = kwargs.get('javascript')
+    react = kwargs.get('react')
+    locale = kwargs.get('locale')
+    plugin_dirs = kwargs.get('plugin_dirs')
+    no_check = kwargs.get('no_check', True)
+
+    if not (python or react or javascript):
+        python = react = javascript = True
+        if babel_cmd == 'compile_catalog':
+            javascript = False
+
+    if not plugin_dirs:
+        _indico_command(babel_cmd, python, javascript, react, locale, no_check)
+    else:
+        for plugin_dir in plugin_dirs:
+            _plugin_command(babel_cmd, python, javascript, react, locale, no_check, plugin_dir)
+
+
+def _make_command(group, cmd_name, babel_cmd, **kwargs):
+    cmd_group = group.command(cmd_name, short_help=f'Perform {babel_cmd} operation on {cmd_name}')
+
+    @_common_translation_options(**kwargs)
+    def wrapper(**kwargs):
+        _command(babel_cmd=babel_cmd, **kwargs)
+
+    return cmd_group(wrapper)
+
+
+_make_command(compile_catalog, 'indico', 'compile_catalog', require_js=False)
+_make_command(extract_messages, 'indico', 'extract_messages')
+_make_command(update_catalog, 'indico', 'update_catalog')
+_make_command(init_catalog, 'indico', 'init_catalog', require_locale=True)
+
+_make_command(compile_catalog, 'plugins', 'compile_catalog', require_js=False,
+              plugins=True)
+_make_command(extract_messages, 'plugins', 'extract_messages', plugins=True)
+_make_command(update_catalog, 'plugins', 'update_catalog', plugins=True)
+_make_command(init_catalog, 'plugins', 'init_catalog', require_locale=True,
+              plugins=True)
+
+_make_command(compile_catalog, 'all-plugins', 'compile_catalog', require_js=False,
+              all_plugins=True)
+_make_command(extract_messages, 'all-plugins', 'extract_messages', all_plugins=True)
+_make_command(update_catalog, 'all-plugins', 'update_catalog', all_plugins=True)
+_make_command(init_catalog, 'all-plugins', 'init_catalog', require_locale=True,
+              all_plugins=True)
+
+
+@cli.command(short_help='Push .pot files to Transifex.')
+def push():
+    """Push .pot files to Transifex."""
+    try:
+        subprocess.run(['tx', 'push', '-s'], check=True)
+    except subprocess.CalledProcessError:
+        click.secho('Error pushing to transifex', fg='red', bold=True, err=True)
+
+
+@cli.command(short_help='Pull translated .po files from Transifex.')
+@click.argument('languages', nargs=-1, required=False)
+def pull(languages):
+    """Pull translations from Transifex.
+
+    Pulls only the language codes specified in `languages` if present.
+    If no `languages`, it pulls all the languages currently available in the translations folder.
+    To pull a new language available on Transifex, simply pass the new language code in the argument.
     """
-    root_path = 'indico/translations'
+    if not languages:
+        languages = [d for d in os.listdir(TRANSLATIONS_DIR) if os.path.isdir(os.path.join(TRANSLATIONS_DIR, d))]
```

Comment:
`en_US` should be skipped since this is not managed via transifex (it's the hardcoded locale). Skipping it avoids a warning when running `tx`.